src/v0/destinations/fb_custom_audience/util.js

```diff
@@ -57,114 +57,104 @@ const getSchemaForEventMappedToDest = (message) => {
 const ensureApplicableFormat = (userProperty, userInformation) => {
   let updatedProperty;
   let userInformationTrimmed;
-  const stringifiedUserInfo = userInformation.toString();
-  switch (userProperty) {
-    case 'EMAIL':
-      updatedProperty = stringifiedUserInfo.trim().toLowerCase();
-      break;
-    case 'PHONE':
-      // remove all non-numerical characters
-      updatedProperty = stringifiedUserInfo.replace(/\D/g, '');
-      // remove all leading zeros
-      updatedProperty = updatedProperty.replace(/^0+/g, '');
-      break;
-    case 'GEN':
-      updatedProperty =
-        stringifiedUserInfo.toLowerCase() === 'f' || stringifiedUserInfo.toLowerCase() === 'female'
-          ? 'f'
-          : 'm';
-      break;
-    case 'DOBY':
-      updatedProperty = stringifiedUserInfo.trim().replace(/\./g, '');
-      break;
-    case 'DOBM':
-    case 'DOBD':
-      userInformationTrimmed = stringifiedUserInfo.replace(/\./g, '');
-      if (stringifiedUserInfo.length < 2) {
-        updatedProperty = `0${userInformationTrimmed}`;
-      } else {
-        updatedProperty = userInformationTrimmed;
-      }
-      break;
-    case 'LN':
-    case 'FN':
-    case 'FI':
-      if (userProperty !== 'FI') {
-        updatedProperty = stringifiedUserInfo.toLowerCase().replace(/[!#$%&@A-Za-z]/g, '');
-      } else {
-        updatedProperty = stringifiedUserInfo.toLowerCase().replace(/[^!#$%&,.?@A-Za-z]/g, '');
-      }
-      break;
-    case 'MADID':
-      updatedProperty = stringifiedUserInfo.toLowerCase();
-      break;
-    case 'COUNTRY':
-      updatedProperty = stringifiedUserInfo.toLowerCase();
-      break;
-    case 'ZIP':
-      userInformationTrimmed = stringifiedUserInfo.replace(/\s/g, '');
-      updatedProperty = userInformationTrimmed.toLowerCase();
-      break;
-    case 'ST':
-    case 'CT':
-      updatedProperty = stringifiedUserInfo
-        .replace(/[^ A-Za-z]/g, '')
-        .replace(/\s/g, '')
-        .toLowerCase();
-      break;
-    case 'EXTERN_ID':
-      updatedProperty = stringifiedUserInfo;
-      break;
-    default:
-      throw new ConfigurationError(`The property ${userProperty} is not supported`);
+  if (isDefinedAndNotNull(userInformation)) {
+    userInformation = userInformation.toString();
```

Comment:
this will throw sonar warning please keep` const stringifiedUserInfo = userInformation.toString();`. It is not allowed to reassign a function parameter as a sonar rule

Reply:
Updated.